src/backend/replication/walproposer.c

```diff
@@ -1527,9 +1529,31 @@ CreateMessageCommitLsnOnly(XLogRecPtr lsn)
 	return msg;
 }
 
+/*
+ * Process all events happened in SS_ACTIVE state, update event set after that.
+ */
+static void
+HandleActiveState(WalKeeper *wk, uint32 events)
+{
+	uint32 newEvents = WL_SOCKET_READABLE;
+
+	if (events & WL_SOCKET_WRITEABLE)
+		if (!SendAppendRequests(wk))
+			return;
+
+	if (events & WL_SOCKET_READABLE)
+		if (!RecvAppendResponses(wk))
+			return;
+
+	if (wk->currMsg != NULL || wk->flushWrite)
```

Comment:
Do we need `wk->currMsg != NULL` condition here, why the latter is not enough?

Reply:
I thought we can have some pending messages here, for example if RecvAppendResponses created new messages without sending them, then if we don't have WL_SOCKET_WRITEABLE in event mask, we won't sent messages anymore.

Now I ran wal_acceptor* tests and it seems that everything is working ok without `wk->currMsg != NULL`, so that's more a paranoid check. But that's still useful if we would want to add messages to queue without immediately sending them.

Reply:
Ok, I don't mind leaving it, though it might worth a comment.